source/developers-guide/customer-streams-extension/index.md

```diff
@@ -0,0 +1,280 @@
+
+```
+<?php
+
+namespace SwagCustomerSearchExtension\Bundle\CustomerSearchBundle;
+
+use Shopware\Bundle\SearchBundle\ConditionInterface;
+
+class ActiveCondition implements ConditionInterface
+{
+    /**
+     * @var bool
+     */
+    protected $active;
+
+    /**
+     * @param bool $active
+     */
+    public function __construct($active)
+    {
+        $this->active = $active;
+    }
+
+    public function getName()
+    {
+        return 'ActiveCondition';
+    }
+
+    public function onlyActive()
+    {
+        return $this->active;
+    }
+}
+```
+This condition describes only on an abstract level what is to be searched for. The actual processing of the condition then happens in the corresponding implementation of the search. Currently the customer search in Shopware is only executed in SQL. Therefore, a handler class is written below, which handles this condition in SQL.
+This is the same concept as in the SearchBundleDBAL and SearchBundleES:
+
+```
+<?php
+
+namespace SwagCustomerSearchExtension\Bundle\CustomerSearchBundleDBAL;
+
+use Shopware\Bundle\CustomerSearchBundleDBAL\ConditionHandlerInterface;
+use Shopware\Bundle\SearchBundle\ConditionInterface;
+use Shopware\Bundle\SearchBundleDBAL\QueryBuilder;
+use SwagCustomerSearchExtension\Bundle\CustomerSearchBundle\ActiveCondition;
+
+class ActiveConditionHandler implements ConditionHandlerInterface
+{
+    public function supports(ConditionInterface $condition)
+    {
+        return $condition instanceof ActiveCondition;
+    }
+
+    public function handle(ConditionInterface $condition, QueryBuilder $query)
+    {
+        $query->andWhere('customer.active = :active');
+
+        /** @var ActiveCondition $condition */
+        $query->setParameter(':active', $condition->onlyActive());
+    }
+}
+```
+The handler can be registered over a compiler tag, named `customer_search.condition_handler`:
+
+```
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="swag_customer_search_extension.active_condition_handler"
+                 class="SwagCustomerSearchExtension\Bundle\CustomerSearchBundleDBAL\ActiveConditionHandler">
+
+            <tag name="customer_search.condition_handler"/>
+        </service>
+
+    </services>
+</container>
+```
+
+## Backend Module extension
+For the condition in the backend to be selected, it is necessary to extend the customer module via ExtJS. The module can be extended over the PostDispatch event of the backend customer controller:
```

Comment:
To support the condition in the backend, it is necessary to extend the customer module via ExtJS. The module can be extended over the PostDispatch event of the backend customer controller: